src/ui/src/js/controllers/admin_garden.js

```diff
@@ -53,15 +53,32 @@ export default function adminGardenController(
     loadGardens();
   };
 
-  EventService.addCallback('admin_system', (event) => {
+  EventService.addCallback('admin_garden', (event) => {
     switch (event.name) {
       case 'GARDEN_CREATED':
-        break;
+        $scope.data.push(event.payload);
+        $apply();
       case 'GARDEN_REMOVED':
-        break;
+        let new_data = []
+        for (var i = 0; i < $scope.data.length; i++) {
+            if ($scope.data[i].id != event.payload.id){
+                new_data.push($scope.data[i]);
+            }
+        }
+        $scope.data = new_data
```

Comment:
This isn't necessary, you can pull an item out of an array using `splice`.

Reply:
Yeah... that works as well. I kept reading up on `slice` instead of `splice` when I first did this. So that is why I avoided the array functions.